### PR TITLE
feat: Add dolby eac3prodec reference decoder

### PR DESCRIPTION
**Pull request by mcesariniflu**
Added Dolby eac3prodec reference decoder.

```
python3 fluster.py run -ts EAC3_ELEMENTARY_STREAMS -d Dolby_EAC3PRODEC_ReferenceDecoder
****************************************************************************************************
Running test suite EAC3_ELEMENTARY_STREAMS with decoder Dolby_EAC3PRODEC_ReferenceDecoder
Using 16 parallel job(s)
****************************************************************************************************

[TEST SUITE             ] (DECODER                          ) TEST VECTOR                              ... RESULT
----------------------------------------------------------------------
[EAC3_ELEMENTARY_STREAMS] (Dolby_EAC3PRODEC_ReferenceDecoder) Audio_ID_special_6ch_640kbps_ddp_joc.ec3 ... Success
[EAC3_ELEMENTARY_STREAMS] (Dolby_EAC3PRODEC_ReferenceDecoder) Audio_ID_2ch_128kbps_ddp.ec3             ... Success
[EAC3_ELEMENTARY_STREAMS] (Dolby_EAC3PRODEC_ReferenceDecoder) Audio_ID_514ch_448kbps_ddp.ec3           ... Success
[EAC3_ELEMENTARY_STREAMS] (Dolby_EAC3PRODEC_ReferenceDecoder) Audio_ID_6ch_256kbps_ddp.ec3             ... Success
[EAC3_ELEMENTARY_STREAMS] (Dolby_EAC3PRODEC_ReferenceDecoder) Audio_ID_6ch_640kbps_ddp_joc.ec3         ... Success
[EAC3_ELEMENTARY_STREAMS] (Dolby_EAC3PRODEC_ReferenceDecoder) Holi_en_6ch_256kbps_ddp.ec3              ... Success
[EAC3_ELEMENTARY_STREAMS] (Dolby_EAC3PRODEC_ReferenceDecoder) Holi_fr_6ch_256kbps_ddp.ec3              ... Success
[EAC3_ELEMENTARY_STREAMS] (Dolby_EAC3PRODEC_ReferenceDecoder) AVSync_2ch_384kbps_25fps_ddp.ec3         ... Success
[EAC3_ELEMENTARY_STREAMS] (Dolby_EAC3PRODEC_ReferenceDecoder) AVSync_2ch_384kbps_2997fps_ddp.ec3       ... Success
[EAC3_ELEMENTARY_STREAMS] (Dolby_EAC3PRODEC_ReferenceDecoder) AVSync_6ch_640kbps_25fps_ddp_joc.ec3     ... Success
[EAC3_ELEMENTARY_STREAMS] (Dolby_EAC3PRODEC_ReferenceDecoder) AVSync_6ch_640kbps_2997fps_ddp_joc.ec3   ... Success
[EAC3_ELEMENTARY_STREAMS] (Dolby_EAC3PRODEC_ReferenceDecoder) Audio_ID_6ch_1024kbps_ddp_joc.ec3        ... Success


Ran 12/12 tests successfully, in 22.453 secs
```

### CODE REVIEW

File: fluster/decoders/dolby_pad.py
Comment (rgonzalezfluendo) [inline fluster/decoders/dolby_pad.py:1]: `eac3_prodec_decoder_reference.py` and `ac4_decoder_reference.py` must be the same file. Like `dolby_pad.py`.

Use `libvpx.py` as reference.



```
class DolbyPADSDecoder(Decoder):
....

@register_decoder
class DolbyEAC3PRODECDecoder(DolbyPADSDecoder):
    """EAC3 reference decoder implementation"""

    codec = Codec.EAC3

@register_decoder
class DolbyAC4DECDecoder(DolbyPADSDecoder):
    """AC4 reference decoder implementation"""

    codec = Codec.AC4
```

**Review by rgonzalezfluendo — DISMISSED** [review]

File: fluster/decoders/dolby_pad.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        input_filepath: str,
+        output_filepath: str,
+        output_format: OutputFormat,
+        timeout: int,
+        verbose: bool,
+        keep_files: bool,
+        optional_params: Optional[Dict[str, Any]] = None,
+    ) -> str:
+        """Decodes input_filepath in output_filepath"""
+        run_command(
```
Comment (rsanchez87) [inline fluster/decoders/dolby_pad.py:56]: Minor suggestion: consider passing -`ddp_decoder_speaker_config` based on `channels_layout` from `optional_params` to ensure the reference decoder output matches the stream's native channel count. 

Although current tests pass without it, it makes the intent explicit and avoids potential mismatches in edge cases

File: fluster/decoders/dolby_pad.py
Comment (rsanchez87) [reply fluster/decoders/dolby_pad.py:1]: Something like:

```
class DolbyPADSDecoder(Decoder):
    binary = "decoder_reference_app_linux_x86_64"
    is_reference = True
    _speaker_config_map: Dict[int, int] = {}

    def decode(self, input_filepath, output_filepath, ..., optional_params=None) -> str:
        cmd = [self.binary, "-i", input_filepath, "-o", output_filepath]
        if self._speaker_config_map and optional_params:
            channels_layout = optional_params.get("channels_layout", 0)
            speaker_config = self._speaker_config_map.get(channels_layout, 1)
            cmd += ["-ddp_decoder_speaker_config", str(speaker_config)]
        run_command(cmd, ...)
        return file_checksum(output_filepath)

@register_decoder
class DolbyAC4DECDecoder(DolbyPADSDecoder):
    codec = Codec.AC4

@register_decoder
class DolbyEAC3PRODECDecoder(DolbyPADSDecoder):
    codec = Codec.EAC3
    _speaker_config_map = {0: 1, 2: 0, 7: 1}
```

Comment (mcesariniflu) [reply fluster/decoders/dolby_pad.py:1]: https://github.com/fluendo/fluster/pull/367#discussion_r3436638764
Ok, that wasn't specified in the original task. 
I'll unify them. Thanks

**Comment by mcesariniflu** [comment]
Unify Dolby AC4 and EAC3prodec reference decoders.

AC4
```
python3 fluster.py run -ts AC4_ELEMENTARY_STREAMS -d DolbyPADSReferenceDecoder
****************************************************************************************************
Running test suite AC4_ELEMENTARY_STREAMS with decoder DolbyPADSReferenceDecoder
Using 16 parallel job(s)
****************************************************************************************************

[TEST SUITE            ] (DECODER                  ) TEST VECTOR                            ... RESULT
----------------------------------------------------------------------
[AC4_ELEMENTARY_STREAMS] (DolbyPADSReferenceDecoder) Audio_ID_2ch_64kbps_25fps_ac4.ac4      ... Success
[AC4_ELEMENTARY_STREAMS] (DolbyPADSReferenceDecoder) Audio_ID_2ch_64kbps_2997fps_ac4.ac4    ... Success
[AC4_ELEMENTARY_STREAMS] (DolbyPADSReferenceDecoder) Audio_ID_6ch_128kbps_25fps_ac4.ac4     ... Success
[AC4_ELEMENTARY_STREAMS] (DolbyPADSReferenceDecoder) Audio_ID_514ch_192kbps_2997fps_ac4.ac4 ... Success
[AC4_ELEMENTARY_STREAMS] (DolbyPADSReferenceDecoder) Audio_ID_6ch_128kbps_2997fps_ac4.ac4   ... Success
[AC4_ELEMENTARY_STREAMS] (DolbyPADSReferenceDecoder) Audio_ID_514ch_192kbps_25fps_ac4.ac4   ... Success
[AC4_ELEMENTARY_STREAMS] (DolbyPADSReferenceDecoder) Holi_en_6ch_128kbps_25fps_ac4.ac4      ... Success
[AC4_ELEMENTARY_STREAMS] (DolbyPADSReferenceDecoder) Holi_fr_6ch_128kbps_2997fps_ac4.ac4    ... Success
[AC4_ELEMENTARY_STREAMS] (DolbyPADSReferenceDecoder) Holi_fr_6ch_128kbps_25fps_ac4.ac4      ... Success
[AC4_ELEMENTARY_STREAMS] (DolbyPADSReferenceDecoder) Holi_en_6ch_128kbps_2997fps_ac4.ac4    ... Success


Ran 10/10 tests successfully, in 8.991 secs

```

EAC3prodec
```
python3 fluster.py run -ts EAC3_ELEMENTARY_STREAMS -d Dolby_EAC3PRODEC_ReferenceDecoder
****************************************************************************************************
Running test suite EAC3_ELEMENTARY_STREAMS with decoder Dolby_EAC3PRODEC_ReferenceDecoder
Using 16 parallel job(s)
****************************************************************************************************

[TEST SUITE             ] (DECODER                          ) TEST VECTOR                              ... RESULT
----------------------------------------------------------------------
[EAC3_ELEMENTARY_STREAMS] (Dolby_EAC3PRODEC_ReferenceDecoder) Audio_ID_special_6ch_640kbps_ddp_joc.ec3 ... Success
[EAC3_ELEMENTARY_STREAMS] (Dolby_EAC3PRODEC_ReferenceDecoder) Audio_ID_2ch_128kbps_ddp.ec3             ... Success
[EAC3_ELEMENTARY_STREAMS] (Dolby_EAC3PRODEC_ReferenceDecoder) Audio_ID_6ch_256kbps_ddp.ec3             ... Success
[EAC3_ELEMENTARY_STREAMS] (Dolby_EAC3PRODEC_ReferenceDecoder) Audio_ID_514ch_448kbps_ddp.ec3           ... Success
[EAC3_ELEMENTARY_STREAMS] (Dolby_EAC3PRODEC_ReferenceDecoder) Audio_ID_6ch_640kbps_ddp_joc.ec3         ... Success
[EAC3_ELEMENTARY_STREAMS] (Dolby_EAC3PRODEC_ReferenceDecoder) Holi_en_6ch_256kbps_ddp.ec3              ... Success
[EAC3_ELEMENTARY_STREAMS] (Dolby_EAC3PRODEC_ReferenceDecoder) Holi_fr_6ch_256kbps_ddp.ec3              ... Success
[EAC3_ELEMENTARY_STREAMS] (Dolby_EAC3PRODEC_ReferenceDecoder) AVSync_2ch_384kbps_25fps_ddp.ec3         ... Success
[EAC3_ELEMENTARY_STREAMS] (Dolby_EAC3PRODEC_ReferenceDecoder) AVSync_2ch_384kbps_2997fps_ddp.ec3       ... Success
[EAC3_ELEMENTARY_STREAMS] (Dolby_EAC3PRODEC_ReferenceDecoder) AVSync_6ch_640kbps_25fps_ddp_joc.ec3     ... Success
[EAC3_ELEMENTARY_STREAMS] (Dolby_EAC3PRODEC_ReferenceDecoder) AVSync_6ch_640kbps_2997fps_ddp_joc.ec3   ... Success
[EAC3_ELEMENTARY_STREAMS] (Dolby_EAC3PRODEC_ReferenceDecoder) Audio_ID_6ch_1024kbps_ddp_joc.ec3        ... Success


Ran 12/12 tests successfully, in 22.132 secs

```

File: fluster/decoders/dolby_pad.py [hunk 2]
```diff
@@ -43,9 +44,38 @@ def decode(
...
+
+@register_decoder
+class EAC3PRODECDecoder(DolbyPADDecoder):
+    """Dolby EC-3 Online Delivery Kit 1.6 elementary stream reference decoder implementation"""
+
+    name = "Dolby_EAC3PRODEC_ReferenceDecoder"
+    description = "Dolby EC-3 Online Delivery Kit 1.6 elementary stream reference decoder implementation"
+    codec = Codec.EAC3
+    # channels_layout 2 -> stereo, 7 (FL+FR+FC+LFE+BL+BR) -> 5.1
+    _speaker_config_map = {2: 2, 7: 6}
```
Comment (rsanchez87) [inline fluster/decoders/dolby_pad.py]: Wrong mapping values. -`ddp_decoder_speaker_config` uses a different numbering than `channels_layout`: **0**=2.0, **1**=5.1, **2**=7.1, **6**=9.1.6. The correct mapping should be:

`_speaker_config_map = {0: 1, 2: 0, 7: 1}`

`channels_layout=2` (2.0) → -`ddp_decoder_speaker_config` 0 (2.0)
`channels_layout=7` (5.1) → -`ddp_decoder_speaker_config` 1 (5.1)
`channels_layout=0` (Auto/JOC) → -`ddp_decoder_speaker_config` 1 (5.1, covered by default)

**Review by rsanchez87 — CHANGES_REQUESTED** [review]

**Comment by rsanchez87** [comment]
Please squash these two commits into one, the first commit adds a file and the second immediately refactors/removes it, so they should be a single atomic commit. 

The final commit message could be:

`feat: add Dolby PADS reference decoder with AC4 and EAC3 support`

File: fluster/decoders/dolby_pad.py (hunk 2 above)
Comment (mcesariniflu) [reply fluster/decoders/dolby_pad.py]: ok

**Review by rsanchez87 — APPROVED** [review]
